wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/RootDirPathProvider.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.project.server.impl;
+
+import com.google.inject.Inject;
+import java.io.File;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+/**
+ * Provides path for projects root directory using: env variable 'CHE_PROJECTS_ROOT' (which is set
+ * by workspace API == "project" volume configured) if not - uses 'che.user.workspaces.storage'
+ * property otherwise - default with '/project' directory (backward comptible solution)
+ *
+ * @author gazarenkov
+ */
+@Singleton
+public class RootDirPathProvider implements Provider<String> {
+
+  private static String DEF = "/projects";
+
+  @Inject(optional = true)
+  @Named("che.user.workspaces.storage")
+  protected File rootFile = null;
+
+  @Override
+  public String get() {
+
+    String path = System.getenv("CHE_PROJECTS_ROOT");
```

Comment:
We can probably benefit of using Guice env properties injection support as it became a common approach throughout a project (e.g. [WorkspaceConfigProvider](https://github.com/eclipse/che/blob/7a52c83f4dbf24f527986550181684f267934387/wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/WorkspaceConfigProvider.java#L61))
```java
@Named("env.CHE_PROJECTS_ROOT") String cheProjectsRoot;
```
What do you think?

Reply:
1 is fine